### PR TITLE
Update hashicorp-vault.md

### LINKED ISSUE #1354: Update HashiCorp Vault secret docs
**What content needs to be created or modified?**
Closing issue 794 to support vaultToken requires an update to docs.

**Describe the solution you'd like**
Update the vault component doc to show support for vaultToken

**Where should the new material be placed?**
https://docs.dapr.io/operations/components/setup-secret-store/supported-secret-stores/hashicorp-vault/

**Additional context**

### PR DESCRIPTION
**Pull request by DarqueWarrior**
Updated the docs to support vaultToken as an alternative to vaultTokenMountPath. This addresses issue #1354

Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**

- [x] [Read the contribution guide](https://docs.dapr.io/contributing/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

This updates the docs to support vaultToken in the Hashicorp Vault secret store.

## Issue reference

PR will close: #1354

### CODE REVIEW

**Comment by orizohar** [comment]
@DarqueWarrior is this in the context of https://github.com/dapr/components-contrib/issues/794? is this documenting a change in the code or is it relevant to v1.0? (currently this doc update is targeting v1.0)

**Comment by DarqueWarrior** [comment]
Correct the change was just made and is not released. Should I close this and target it somewhere else?

**Comment by AaronCrawfis** [comment]
> Correct the change was just made and is not released. Should I close this and target it somewhere else?

Yes please. If it was part of last week's v1.1 release re-target to the v1.1 branch. If it's still unreleased target the v1.2 branch

**Comment by DarqueWarrior** [comment]
Ok I moved to 1.2.

File: daprdocs/content/en/operations/components/setup-secret-store/supported-secret-stores/hashicorp-vault.md [hunk 1]
```diff
@@ -51,6 +53,7 @@ The above example uses secrets as plain strings. It is recommended to use a loca
 | skipVerify | N | Skip TLS verification. Defaults to `"false"` | `"true"`, `"false"` |
 | tlsServerName | N | TLS config server name | `"tls-server"` | 
 | vaultTokenMountPath | Y | Path to file containing token | `"path/to/file"` |
+| vaultToken | Y | Token | `"tokenValue"` |
```
Comment (AaronCrawfis) [inline daprdocs/content/en/operations/components/setup-secret-store/supported-secret-stores/hashicorp-vault.md]:
```suggestion
| vaultToken | Y | [Token](https://learn.hashicorp.com/tutorials/vault/tokens) for authentication within Vault.  | `"tokenValue"` |
```

Adding a little more to the description

**Review by AaronCrawfis — APPROVED** [review]

**Pull request merged**